src/libraries/System.Runtime/tests/System/GCTests.cs

```diff
@@ -1017,23 +1014,21 @@ private static void AllocateArrayNegativeSize(int negValue)
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/33583", TestRuntimes.Mono)]
+        [SkipOnMono("Mono can allocate arrays of int.MaxValue length")]
```

Comment:
```suggestion
        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotIntMaxValueArrayIndexSupported))]
```

We try to avoid using SkipOnMono

Reply:
We still have quite a few existing SkipOnMono (https://github.com/dotnet/runtime/search?q=SkipOnMono&unscoped_q=SkipOnMono) so I'd be fine with addressing this in a follow-up PR to avoid redoing CI (it seems a bit flaky today).